presto-main/src/main/java/com/facebook/presto/operator/exchange/LocalExchange.java

```diff
@@ -173,7 +173,7 @@ private static PartitionFunction createPartitionFunction(
         ConnectorNodePartitioningProvider partitioningProvider = partitioningProviderManager.getPartitioningProvider(partitioning.getConnectorId().get());
         ConnectorBucketNodeMap connectorBucketNodeMap = partitioningProvider.getBucketNodeMap(
                 partitioning.getTransactionHandle().orElse(null),
-                session.toConnectorSession(),
+                session.toConnectorSession(partitioning.getConnectorId().get()),
```

Comment:
why the original code has null pointer issue? Maybe explain it a bit in commit message?

Reply:
``toConnectorSession()`` will call
``new FullConnectorSession(this, identity.toConnectorIdentity());``

public FullConnectorSession(Session session, ConnectorIdentity identity)
    {
        this.session = requireNonNull(session, "session is null");
        this.identity = requireNonNull(identity, "identity is null");
        this.properties = null;
        this.connectorId = null;
        this.catalog = null;
        this.sessionPropertyManager = null;
        this.sqlFunctionProperties = session.getSqlFunctionProperties();
    }

here will set properties/connecotId/catalog/sessionPropertyManager null

hence we should call ``ConnectorSession toConnectorSession(ConnectorId connectorId)``
public ConnectorSession toConnectorSession(ConnectorId connectorId)
    {
        requireNonNull(connectorId, "connectorId is null");

        return new FullConnectorSession(
                this,
                identity.toConnectorIdentity(connectorId.getCatalogName()),
                connectorProperties.getOrDefault(connectorId, ImmutableMap.of()),
                connectorId,
                connectorId.getCatalogName(),
                sessionPropertyManager);
    }